### ISSUE TITLE
[Feature : Map Location] Icons

### ISSUE THREAD

**Issue by wolfgang-ch** [mytourbook/mytourbook#1318]
### What is missing ?

There are 4 different map locations:

| Icon | Description|
|---|---|
|![map-location-marker](https://github.com/mytourbook/mytourbook/assets/1283445/b862ec97-7400-4705-9eba-ddfdc0420f6d)| Tour location<br>|
| ![map-location-marker-start](https://github.com/mytourbook/mytourbook/assets/1283445/cb8ec82d-0659-4f7a-be62-935c0e29eb54) | Tour start location |
| ![map-location-marker-end](https://github.com/mytourbook/mytourbook/assets/1283445/fa7379cf-dc61-48cf-bbb2-032e88c33bf4) | Tour end location |
| ![map-location-marker-address](https://github.com/mytourbook/mytourbook/assets/1283445/64925a7a-4aae-4405-9958-8c59fbce01af) | Address location <br>Is **not** associated with a tour, it's like a map bookmark|

**Comment by wolfgang-ch** [comment]
In the last part of the above video (which was just added), the map is panned but the moved map locations do not stick to the map image or vice versa. 

So the whole 2D map marker and map locations needs an overhaul that panning the map is more smoothly. This will be a larger project.

**Comment by FJBDev** [comment]
> So the whole 2D map marker and map locations needs an overhaul that panning the map is more smoothly. This will be a larger project.

Maybe it's time to use a 2d map framework to not "reinvent the wheel"
Have you considered modern projects such as the below ones :
- [ArcGIS maps](https://developers.arcgis.com/java/maps-2d/tutorials/display-a-web-map/#open-a-java-project-with-gradle)
- [SWT Map Widget](https://mappanel.sourceforge.net/swt/)
- [Apache SIS](https://sis.apache.org) - Library for developing geospatial applications.
- [Geo](https://github.com/davidmoten/geo) - GeoHash utilities in Java.
- [GeoTools](https://geotools.org) - Library that provides tools for geospatial data. (LGPL-2.1-only)
- [GraphHopper](https://github.com/graphhopper/graphhopper) - Road-routing engine. Used as a Java library or standalone web service.
- [H2GIS](http://www.h2gis.org) - Spatial extension of the H2 database. (LGPL-3.0-only)
- [Jgeohash](https://astrapi69.github.io/jgeohash/) - Library for using the GeoHash algorithm.
- [Mapsforge](https://github.com/mapsforge/mapsforge) - Map rendering based on OpenStreetMap data. (LGPL-3.0-only)
- [Spatial4j](https://github.com/locationtech/spatial4j) - General-purpose spatial/geospatial library.

**Comment by wolfgang-ch** [comment]
The 2D map is not only just a map, it has many features which an alternative also should have, or it must be also reinvented, e.g.

* Trackcolors depending on the dataserie value
* Paint photos
* https://mytourbook.sourceforge.io/mytourbook/index.php/documentation/show-tours/tour-map

The map itself performs very good it's only some additional features which have issues. I would like to use an existing 2D map framework but it would need a lot of work to have the same features as the current map. 

The 2D map is already based on Geoclipse map https://sourceforge.net/projects/geoclipse/

Here are a few performance data of the map without any additional features for my 2560x1440 screen

```
Painting the map image on the screen

08:02:16.296'447 onPaint() -   1.767 ms
08:02:16.484'166 onPaint() -   1.754 ms
08:02:18.055'365 onPaint() -   1.240 ms
08:02:22.018'053 onPaint() -   1.288 ms

Painting all tile images into the map image

08:02:48.518'651 paint_10_PaintMapImage()  -   3.327 ms
08:02:48.538'203 paint_10_PaintMapImage()  -   3.283 ms
08:02:48.570'554 paint_10_PaintMapImage()  -   3.244 ms

```

**Comment by wolfgang-ch** [comment]
With painting the markers in another thread, the markers (and also other drawings) do now stick to the map

https://github.com/mytourbook/mytourbook/assets/1283445/38264244-6c66-4124-865b-5a773ee47297

**Comment by wolfgang-ch** [comment]
After using a separate thread for drawing the tour markers, new features can be implemented which was not possible with the last marker paintings. 

Now the markers can be clustered by using the same clustering algorithm as in the 2.5D map

Don't be irritated that the labels a changing constantly, this is just because they are scrambled which normally is not enabled.


https://github.com/mytourbook/mytourbook/assets/1283445/2925f6fc-21a9-44be-9f2b-f4a46c32415f

**Comment by FJBDev** [comment]
> The 2D map is not only just a map, it has many features which an alternative also should have, or it must be also reinvented, e.g.
> 
>     * Trackcolors depending on the dataserie value
> 
>     * Paint photos
> 
>     * https://mytourbook.sourceforge.io/mytourbook/index.php/documentation/show-tours/tour-map
> 
> 
> The map itself performs very good it's only some additional features which have issues. I would like to use an existing 2D map framework but it would need a lot of work to have the same features as the current map.
> 
> The 2D map is already based on Geoclipse map https://sourceforge.net/projects/geoclipse/
> 
> Here are a few performance data of the map without any additional features for my 2560x1440 screen
> 
> ```
> Painting the map image on the screen
> 
> 08:02:16.296'447 onPaint() -   1.767 ms
> 08:02:16.484'166 onPaint() -   1.754 ms
> 08:02:18.055'365 onPaint() -   1.240 ms
> 08:02:22.018'053 onPaint() -   1.288 ms
> 
> Painting all tile images into the map image
> 
> 08:02:48.518'651 paint_10_PaintMapImage()  -   3.327 ms
> 08:02:48.538'203 paint_10_PaintMapImage()  -   3.283 ms
> 08:02:48.570'554 paint_10_PaintMapImage()  -   3.244 ms
> ```

I understand it would be a big overhaul to use a different SDK/framework but looking at the [Geoclipse website](https://sourceforge.net/projects/geoclipse/), it hasn't been modified since 2013, which is **11 years ago**, it seems like it's a dead project.

Maybe another modern framework doesn't support all the current 2D Map features but on the long run, maybe at some point, it will be better to invest into a maintained framework.
My favorite one is [ArcGIS ](https://developers.arcgis.com/java/maps-2d/tutorials/display-a-web-map/)and it seems like it supports waypoints....

**Comment by wolfgang-ch** [comment]
ArcGIS has a complicated pricing structure https://developers.arcgis.com/pricing/

**Comment by wolfgang-ch** [comment]
> looking at the [Geoclipse website](https://sourceforge.net/projects/geoclipse/), it hasn't been modified since 2013, which is **11 years ago**, it seems like it's a dead project.

In the beginning I was happy to use a map which worked in an Eclipse RCP app but MT do not depend on geoclipse anymore, it is now integrated and was used as a basis and enhanced in many areas.

After finding a solution to draw the map in the background https://github.com/mytourbook/mytourbook/issues/1318#issuecomment-2061531460 I'm very happy to re-implement some issues https://github.com/mytourbook/mytourbook/issues/1323

However there is one thing which is currently not the best, it's the transparency of the image. Currently it's just one color and not the whole color spectrum. I already used the whole color spectrum when painting map profiles (a map tile has more than one map provider) but this is very complex. So for now I will use the 1 color transparency.

In the beginning I also looked for a chart framework but was not happy which was on the "market", so I created my own.
